<launch>
    <!-- Parameters -->
    <arg name="target_num" default="3"/>
    <arg name="win_size" default="40"/>
    <arg name="pred_win_size" default="20"/>
    <arg name="pred_model_path" default="$(find rotors_gazebo)/pred_model_ckpt/"/>
    <arg name="logging" default="true"/>

    <!-- Drone 1 Processor -->
    <node pkg="rotors_gazebo" type="drone_bbox_node.py" name="drone1_processor" output="screen">
        <param name="drone_id" value="1"/>
        <param name="target_num" value="$(arg target_num)"/>
        <param name="win_size" value="$(arg win_size)"/>
        <param name="pred_win_size" value="$(arg pred_win_size)"/>
        <param name="pred_model_path" value="$(arg pred_model_path)"/>
        <param name="log_path" value="$(find rotors_gazebo)/drone_1_logs/"/>
        <param name="logging" value="$(arg logging)"/>
    </node>
    
    <!-- Drone 2 Processor -->
    <node pkg="rotors_gazebo" type="drone_bbox_node.py" name="drone2_processor" output="screen">
        <param name="drone_id" value="2"/>
        <param name="target_num" value="$(arg target_num)"/>
        <param name="win_size" value="$(arg win_size)"/>
        <param name="pred_win_size" value="$(arg pred_win_size)"/>
        <param name="pred_model_path" value="$(arg pred_model_path)"/>
        <param name="log_path" value="$(find rotors_gazebo)/drone_2_logs/"/>
        <param name="logging" value="$(arg logging)"/>
    </node>
    
    <!-- Drone 3 Processor -->
    <node pkg="rotors_gazebo" type="drone_bbox_node.py" name="drone3_processor" output="screen">
        <param name="drone_id" value="3"/>
        <param name="target_num" value="$(arg target_num)"/>
        <param name="win_size" value="$(arg win_size)"/>
        <param name="pred_win_size" value="$(arg pred_win_size)"/>
        <param name="pred_model_path" value="$(arg pred_model_path)"/>
        <param name="log_path" value="$(find rotors_gazebo)/drone_3_logs/"/>
        <param name="logging" value="$(arg logging)"/>
    </node>

</launch>





